app/services/HCAService.js

```diff
@@ -0,0 +1,264 @@
+import Yaml from 'js-yaml';
+import PushNotification from 'react-native-push-notification';
+import RNFetchBlob from 'rn-fetch-blob';
+
+import { AUTHORITIES_LIST_URL } from '../constants/authorities';
+import {
+  AUTHORITY_SOURCE_SETTINGS,
+  HCA_AUTO_SUBSCRIPTION,
+} from '../constants/storage';
+import { GetStoreData, SetStoreData } from '../helpers/General';
+import languages from '../locales/languages';
+import { LocationData } from './LocationService';
+
+/**
+ * Singleton class to interact with health care authority data
+ */
+class HCAService {
+  /**
+   * Fetches the raw YAML file containing a list of all
+   * of the registered Health Care Authorities.
+   * Saves the response as a cached file for performance.
+   * @returns {void}
+   */
+  async fetchAuthoritiesYaml() {
+    return await RNFetchBlob.config({
```

Comment:
omg if this can access local files, it's probably what I needed for eula.html reading!

Reply:
So this was code I copy/pasted - but I believe it's taking the response and caching it as a file, not accessing that cached file.

https://github.com/joltup/rn-fetch-blob#download-to-storage-directly